### DISCUSSION TITLE
Add ability to access downloaded files and read them into InputStream via API call to Selenium Host

### DISCUSSION THREAD

**Discussion by dmitry-minchuk** [SeleniumHQ/docker-selenium#2322] [Ideas]
This is already done in Selenoid: [Link](https://aerokube.com/selenoid/latest/#_accessing_files_downloaded_with_browser)
Why not to do the same for dockerized Selenium/Seleniarm?

We will have to just build a correct URL like that:
`GET http://selenium-host:4444/download/f2bcd32b-d932-4cdc-a639-687ab8e4f840/myfile.txt`

There are numerous advantages of such approach:
1. No mess with file management - all the downloaded files will be wiped with the browser container
2. No file system access permissions issues
3. No issues with accessing file system when server instances are provisioned automatically and configuration in that case may be tricky

Java code in this case looks simple:

>     private static InputStream download(String fileName) {
>         String url = String.format(fullMethodUrl, BrowserController.getSessionId().get(), fileName);
>         InputStream file = null;
>         HttpRequest request = HttpRequest
>                 .newBuilder()
>                 .uri(URI.create(url))
>                 .GET()
>                 .build();
>         HttpResponse<InputStream> response;
>         try {
>             response = HttpClient
>                     .newHttpClient()
>                     .send(request, HttpResponse.BodyHandlers.ofInputStream());
>         } catch (IOException | InterruptedException e) {
>             e.printStackTrace();
>             return null;
>         }
>         file = response.body();
>         return file;
>     }

**Comment by VietND96** [comment]
Access download file via client bindings, I think you can refer to this https://www.selenium.dev/documentation/webdriver/drivers/remote_webdriver/#downloads

**Comment by VietND96** [comment]
For the API access contents in container, I think we will solve it once this feat https://github.com/SeleniumHQ/docker-selenium/issues/1626 able to implement
